### ISSUE TITLE
Regression on Select with IE11

### ISSUE THREAD

**Issue by harbalk** [ibm-js/deliteful#546]
When running the samples/Select.html on IE11, I found a little problem when selecting an object.
The object is selected, but the text describing it doesn't appear anymore in the text field.

It works fine on deliteful 0.6.

**Comment by cjolif** [comment]
In @AdrianVasiliu absence would you mind digging a little bit in the problem?

**Comment by harbalk** [comment]
I have found the commit who introduce this regression : 
88d542b00c9025dfa6678abe45e84ffd814eb68a
I don't know why but on IE11 the code doesn't pass in the following if :
```
if (!this._duringInteractiveSelection) {
	this.notifyCurrentValue("renderItems");
}
```
@AdrianVasiliu what do you think about it ?

**Comment by AdrianVasiliu** [comment]
@harbalk, I'm not sure why you think 
```
this.notifyCurrentValue("renderItems");
````
should execute in this case. This is about interactive selection (vs programmatic), and the fact the flag `_duringInteractiveSelection` is at `true` in expected. The notifyCurrentValue() call doesn't execute on other browsers either (where it works fine). So I guess the trouble isn't precisely the fact that this line doesn't execute on IE11 (again, it doesn't execute on other browsers either). When I'll have a bit of time I'll try to dig into it, but of course if you can find something on your side it would be welcome.

**Comment by harbalk** [comment]
Oh yes you're right excuse me the function is not executed in the other browsers.
However excuting this.notifyCurrentValue("renderItems") solve the problem if you do the test...

**Comment by AdrianVasiliu** [comment]
It does solve the problem but there's a reason behind not executing it for interactive selection (explained in the comment above this piece of code), so we need to understand what exactly goes wrong on IE11 to find the correct fix.

**Comment by AdrianVasiliu** [comment]
It seems the issue is caused by IE's weird behavior when setting the `value` of an HTML `select` while its options do not have a `value` attribute. This seems to hold for IE 9, 10 and 11 (tested with IE11 and emulation of IE10 and IE9). 
I reproduced it in pure HTML here: http://jsfiddle.net/adrian_vasiliu/3x32dopL/. This contains 3 `<select>`. The first 2 do not have a `value` attribute on `<option>`, the 3rd has it. For the first select, select.selectedIndex is set: works fine. For the second, the `value` property is set: it works in Chrome/FF, but not in IE. For the third, the `value` property is set, and it works fine including in IE.

As far as I can see, this behavior in IE isn't correct, according to: 
* http://www.w3.org/TR/html5/forms.html#htmlselectelement says that select's `value` can be changed (is not readonly).
* http://www.w3.org/TR/html5/forms.html#the-option-element says that option's `value` defaults to its `text` attribute: "The value of an option element is the value of the value content attribute, if there is one, or, if there is not, the value of the element's text IDL attribute".

This explains why deliteful/Select works fine in IE in tests/functional/Select.html: differently than in samples/Select.html, the options here have a `value` attribute.

I'll create a quick PR for working it around.

**Comment by AdrianVasiliu** [comment]
Just tell if I can merge the PR #558.

**Comment by cjolif** [comment]
master is open for fixes. So feel free to merge. In any case thanks a lot of the ifx.

**Comment by AdrianVasiliu** [comment]
Ok, thanks, will do.

If wondering why this IE issue didn't hurt before my commit 88d542b00c9025dfa6678abe45e84ffd814eb68a: because before this commit deliteful/Select used to (unnecessarily) refill the underlying HTML select. Thus the issue was hidden, but it also produced the scroll jump.

**Comment by AdrianVasiliu** [comment]
Fixed via 6a2ac4e6ea50bde9d66a58daab9c78f050740c9a.
